### ISSUE TITLE
Add support for properties files

### ISSUE THREAD

**Issue by jessechahal** [mozilla/sops#158]
Properties files are still used heavily in the Jvm world (which means java, scala, and tons of other languages). It would be nice if there was support for encrypting the values within a properties file. Property files are typically much simpler then yaml or json files. Every entry is a key/value pair separated by an `=` symbol. There are no dictionaries, lists, or any other type of tree structures within a properties file. 

I haven't gone through the code base yet but how difficult do people believe it would be to add support for this filetype (and possibly other similar filetypes with different whitespace and seperate rules?)

**Comment by autrilla** [comment]
This should be relatively easy to do. Basically all you have to do is comply with [this interface](https://github.com/mozilla/sops/blob/master/sops.go#L244-L250). You can see how we do this for json [here](https://github.com/mozilla/sops/blob/master/json/store.go#L183-L256).

The main difference here is that as you said .properties files don't have maps or sequences, and sops's [`Metadata`](https://github.com/mozilla/sops/blob/master/sops.go#L219-L226) is something like a map, and it contains a sequence. For this reason marshaling and unmarshaling `Metadata` is going to be the most complex part. Otherwise, everything else can be parsed onto a simple `TreeBranch`, which is just a slice (array) of `TreeItem`s, which are simple key-values.

If you decide to give it a shot, don't hesitate to ping me for questions.

**Comment by jessechahal** [comment]
I haven't looked into golang yet (python and java guy) but this seems like a good place to start looking into it. Any good resources you would suggest I look into regarding golang?

**Comment by autrilla** [comment]
I just used the official tour at https://tour.golang.org/. I don't know of any better resources to learn.

**Comment by jessechahal** [comment]
Is there a contributors guide? I'm already seeing problems with my fork not being able to build binaries even without any changes. Have others forked the golang version and attempted to build and test with it? As someone who has never done golang development before having to figure out things like the dependency management is done with govend (which I only found luckily through some old commit messages) and that the vendor for this project is go.mozilla.org instead of github.com/mozilla makes the project difficult to contribute to even if I was an experienced golang programmer.

**Comment by autrilla** [comment]
I will create a CONTRIBUTING.md on Monday. Sorry for the inconvenience, @jessechahal

**Comment by jessechahal** [comment]
thanks, I'll take a look during the week to see if I can get properties files working. I'll probably just spend a bit of time today learning golang instead.

**Comment by magthe** [comment]
How's this progressing?

I'm guessing not at all at the moment :(

**Comment by autrilla** [comment]
Yeah, this isn't really progressing. Part of it is on me for not writing the contribution guide (#170). I'll see if I can get around to doing that eventually...

**Comment by jvehent** [comment]
As the maintainers of Sops, the open source project, we try to support contributors who want specific features as much as possible, but as engineers with busy schedules, we only implement the features we directly need. That is to say, it's unlikely we'll implement this ourselves, as we don't use property files, but we'd be happy to help someone do it.

**Comment by andreas-bradahl** [comment]
Guess there's no update on this? Would really help our organization, as all our applications use .properties files.

**Comment by autrilla** [comment]
There's no update. If it's useful to your organization, maybe they can contribute the time to implement it?

**Comment by shay-berman** [comment]
we also need  .properties file support.
But actually I saw that if you add .env suffix to sops, then it encrypt properties files well (instead of raw text it encrypt it by property. So i think its quite enough.
Thoughts?
